src/hotspot/share/opto/superword.cpp

```diff
@@ -2540,9 +2550,9 @@ void VLoopTypes::compute_vector_element_type() {
             // be vectorized if the higher order bits info is imprecise.
             const Type* vt = vtn;
             int op = in->Opcode();
-            if (VectorNode::is_shift_opcode(op) || op == Op_AbsI || op == Op_ReverseBytesI) {
+            if (!can_subword_truncate(in)) {
```

Comment:
It seems `can_subword_truncate` does not cover `VectorNode::is_shift_opcode`, is that correct? Maybe we are missing IR tests that catch this, scary!

Reply:
In this case since the condition is negated, the old condition should still be true since it falls through to the `return false` at the end of the function. Previously, it checked for a small list of nodes as requiring truncation handling which allows nodes that were not whitelisted to slip through and produce incorrect code. Now, we check for a larger group of nodes that we know do not need handling for truncation, so that any nodes not whitelisted will still produce valid code, but will not vectorize (until #23413).